centreon/tests/e2e/features/Dashboards/11-widget-top-bottom-configuration/index.ts

```diff
@@ -347,13 +347,20 @@ When(
 Then(
   'the displayed value format for this metric has been updated from human-readable to exhaustive',
   () => {
-    cy.get('.visx-group text:first-child')
-      .invoke('text')
-      .then((text) => {
-        if (parseFloat(text) !== 0) {
-          expect(text).to.match(/\d+\.\d{3,}/);
-        }
-      });
+    cy.waitUntil(
+      () =>
+        cy
+          .get('.visx-group text:first-child')
+          .invoke('text')
+          .should((text) => {
+            const metricRegex = /\d+\.\d{3,}/;
+            const floatValue = parseFloat(text);
+            expect(floatValue).not.to.equal(0);
```

Comment:
according to [waitUntil documentation](https://github.com/NoriSte/cypress-wait-until?tab=readme-ov-file#usage), you must not add assertions, otherwise it will throw an exception and exit :
```
you cannot put assertions inside checkFunction.
There is no way to avoid a test failure if an assertion throws an error.
You must manually check what the assertions would check for you.
The most common case is checking that an element exists or not, instead of using cy.get('#id').should('exist') you should check that Cypress.$('#id').length is greater than 0
```
```suggestion
```